daemon/audit.c

```diff
@@ -684,18 +684,16 @@ audit_kernel_handle_log(int fd, UNUSED unsigned events, UNUSED event_io_t *io, v
 		log_record = NLMSG_DATA(nlmsg);
 		int uid = -1;
 		int pid = -1;
-		char res[128];
-		char *record_msg = mem_new0(char, MAX_AUDIT_MESSAGE_LENGTH);
-		sscanf(log_record, "%*s pid=%d uid=%d %*s %*s msg='%[^']'", &pid, &uid, record_msg);
-		if (sscanf(record_msg, "%*[^'] res=%s'", res)) {
-			container_t *c = cmld_container_get_by_uid(uid);
-			c = c ? c : cmld_containers_get_a0();
-			char *record_type = mem_printf("type=%d", type);
-			audit_log_event(container_get_uuid(c), strcmp(res, "success") ? FSA : SSA,
-					CMLD, KAUDIT, record_type,
-					uuid_string(container_get_uuid(c)), 2, "msg", log_record);
-			mem_free(record_type);
-		}
+		sscanf(log_record, "%*s pid=%d uid=%d %*8970c", &pid, &uid);
+		char *res = strstr(log_record, "res=") + 4;
```

Comment:
return value of strstr should be checked to avoid using a NULL pointer

Reply:
done